File: practice-01/exercise-04/c/main.c
#include <stdio.h>
#define N 4
#define M 20

struct Candidate
{
  char name[M];
  char genre;
  int dni;
  int age;
  int code;
};

void load(struct Candidate x[N])
{
  for (int i = 0; i < N; i++)
  {
    printf("Nombre: ");
    fgets(x[i].name, sizeof(char) * M, stdin);
    printf("Genero: ");
    scanf("%c", &x[i].genre);
    printf("DNI: ");
    scanf("%d", &x[i].dni);
    printf("Edad: ");
    scanf("%d", &x[i].age);
    printf("Codigo: ");
    scanf("%d", &x[i].code);

    while (getchar() != '\n')
      ;
    printf("\n");
  }
}

void womenCounter(struct Candidate x[N])
{
  int count = 0;
  for (int i = 0; i < N; i++)
  {
    if (x[i].genre == 'F' && x[i].age < 30)
    {
      count++;
    }
  }
  printf("Cantidad de mujeres: %d", count);
}

void averageMenAge(struct Candidate x[N])
{
  int count = 0;
  int sum = 0;
  for (int i = 0; i < N; i++)
  {
    if (x[i].genre == 'M')
    {
      sum += x[i].age;
      count++;
    }
  }
  printf("Edad promedio de varones: %.2f", (float)(sum / count));
}

void printCandidates(struct Candidate x[N])
{
  for (int i = 0; i < N; i++)
  {
    if (x[i].code == 14)
    {
      printf("candidato: %s, dni:%d\n", x[i].name, x[i].dni);
    }
  }
}

void main()
{
  struct Candidate candidates[N];
  load(candidates);
  printf("\n");
  womenCounter(candidates);
  printf("\n");
  averageMenAge(candidates);
  printf("\n");
  printCandidates(candidates);
}

File: practice-01/exercise-04/go/main.go
package main

import "fmt"

const N = 4

type Candidate struct {
	name  string
	genre string
	dni   int
	age   int
	code  int
}

func load(x *[N]Candidate) {
	for i := 0; i < N; i++ {
		fmt.Print("Nombre: ")
		fmt.Scan(&x[i].name)
		fmt.Print("Genero: ")
		fmt.Scan(&x[i].genre)
		fmt.Print("DNI: ")
		fmt.Scan(&x[i].dni)
		fmt.Print("Age: ")
		fmt.Scan(&x[i].age)
		fmt.Print("Code: ")
		fmt.Scan(&x[i].code)
	}
}

func countWomen(x *[N]Candidate) {
	var count = 0
	for _, c := range x {
		if c.genre == "F" && c.age < 30 {
			count++
		}
	}
	fmt.Printf("Cantidad de mujeres -30 años: %d\n", count)
}

func averageMenAge(x *[N]Candidate) {
	var count = 0
	var sum = 0
	for _, c := range x {
		if c.genre == "M" {
			sum += c.age
			count++
		}
	}
	fmt.Printf("Edad promedio de los varones: %.2f\n", float32(sum/count))
}

func printCandidates(x *[N]Candidate) {
	for _, c := range x {
		if c.code == 14 {
			fmt.Printf("nombre: %s, dni: %d\n", c.name, c.dni)
		}
	}
}

func main(){
	var candidate [N]Candidate
	load(&candidate)
	fmt.Println()
	countWomen(&candidate)
	fmt.Println()
	averageMenAge(&candidate)
	fmt.Println()
	printCandidates(&candidate)
}
